src/coreclr/jit/morph.cpp

```diff
@@ -12056,6 +12072,23 @@ GenTree* Compiler::fgMorphSmpOp(GenTree* tree, MorphAddrContext* mac)
                 return fgMorphSmpOp(tree, mac);
             }
 
+            // -a / C => a / -C, where C is constant
+            // DIV(NEG(a), C) => DIV(a, NEG(C))
+            if (fgGlobalMorph && op1->OperIs(GT_NEG) && !op1->gtGetOp1()->IsCnsIntOrI() && op2->IsCnsIntOrI() &&
+                !op2->IsIconHandle() && op2->AsIntCon()->IconValue() != 1 && op2->AsIntCon()->IconValue() != 0 &&
+                op2->AsIntCon()->IconValue() != -1)
+            {
+                // tree: DIV
+                // op1: a
+                // op2: NEG
+                // op2Child: C
+
+                tree->AsOp()->gtOp1 = op1->gtGetOp1();
+                DEBUG_DESTROY_NODE(op1);
+                op2->AsIntCon()->SetIconValue(-op2->AsIntCon()->IconValue()); // -C
+                return tree;
```

Comment:
I think this is the cause of the problem. You return the tree immediately but if you see after the switch-case, we update the tree flags. In your case, when we hit the assert, because `GTF_CALL` is not yet removed from the node because here, we return immediately. I believe same applies to the `GT_MUL` case.

I am not too familiar with the morph and probably @sandreenko  should confirm.